pkg/common/providers/core/provider.go

```diff
@@ -31,4 +31,5 @@ type ConfigProvider func() ([]ConfigBackend, error)
 //ConfigBackend backend for all config types in SDK
 type ConfigBackend interface {
 	Lookup(key string) (interface{}, bool)
+	Set(key string, value interface{})
```

Comment:
This interface is intended for the SDK to read from a ConfigBackend.
As the SDK itself has no need to write to the ConfigBackend, this interface does not need a Set.

Additionally I am concerned about the possibility of breaking some downstream users who might have built to the existing interface.

Reply:
Yes, SDK doesn't need this method, but SDK users do. For example, when some channels added to HLF, I have to edit the connection profile and restart the services. This is not very convenient for the SDK user.

Reply:
fab.EvenService doesn't work with channels that not specified in connection profile (and provided by ConfigBackend). So we need the ability to change the ConfigBackend on the fly.

Reply:
The interface that the SDK consumes doesn't need to be changed for you to have a ConfigBackend that is settable.

Reply:
@troyronda how? config.FromFile returns ConfigProvider, which returns an array of interfaces, not structures:
`type ConfigProvider func() ([]ConfigBackend, error)`

Thus, I have no way to edit the config backend.

Reply:
How about injecting a mutable `ConfigProvider` when you instantiate the SDK with `New`, and keep the reference to that instance?

https://github.com/hyperledger/fabric-sdk-go/blob/7bf58790be693b74f78e37393781988244b0290b/pkg/fabsdk/fabsdk.go#L75

Reply:
How about injecting a mutable ConfigProvider when you instantiate the SDK with `New`, and keep the reference to that instance?

Reply:
Not sure I understand what that means in practice. Easier to use custom endpoint config to solve this problem.

Reply:
Hi, I have similar concerns, I have a similar use case with dynamic channels being created. How do we injecting a mutable ConfigProvider? Some examples or resources would be helpful thank you.

Reply:
I don't understand why do you need to change the interface here. What do you mean with 'connection profile' ?